guides/md/writing_quantization_compatible_layers.md

```diff
@@ -0,0 +1,612 @@
+# Writing Quantization-Compatible Layers in Keras
+
+**Author:** [Jyotinder Singh](https://x.com/Jyotinder_Singh)<br>
+**Date created:** 2025/10/16<br>
+**Last modified:** 2025/10/16<br>
+**Description:** Complete guide for writing quantization-compatible Keras layers.
+
+
+<img class="k-inline-icon" src="https://colab.research.google.com/img/colab_favicon.ico"/> [**View in Colab**](https://colab.research.google.com/github/keras-team/keras-io/blob/master/guides/ipynb/writing_quantization_compatible_layers.ipynb)  <span class="k-dot">•</span><img class="k-inline-icon" src="https://github.com/favicon.ico"/> [**GitHub source**](https://github.com/keras-team/keras-io/blob/master/guides/writing_quantization_compatible_layers.py)
+
+
+
+---
+## What are quantization-compatible layers?
+
+Keras lets you optimize models via post-training quantization (PTQ) by calling
+the `layer.quantize(...)` or `model.quantize(...)` APIs. Keras exposes an
+extensible framework for defining quantization-compatible layers. This lets you
+author custom layers that plug into the quantization framework, can be quantized
+to INT8 or INT4, and saved/loaded with quantization metadata.
+
+A quantization-compatible layer needs to implement a few hooks, so that it can:
+
+- Switch its variables to quantized representations.
+- Use a quantization-aware forward path at inference.
+- Save and load quantization metadata with the model.
+
+In this guide, we'll implement a simple layer that supports INT8 PTQ. The same
+patterns generalize to INT4 quantization and FP8 mixed-precision training.
```

Comment:
Later in the guide, you explain how to do INT4, and in particular you mention `_int4_build` the `_int4_call`.

However, you never talk about FP8 anymore, and in particular, the names `_float8_build` and `_float8_call` are not mentioned. I think you should at least mention those 2 method names. Initially I thought a table format might be good to show FP8 / INT8 / INT4 but I'll leave it up to you.

Reply:
FP8 is a mixed precision training technique that happens to do a quantization step in the middle. It includes more sophisticated variable history tracking and fake-quant injection which might be irrelevant to the particular audience of this guide.